baybe/recommenders/pure/bayesian/botorch.py

```diff
@@ -105,16 +105,17 @@ def _recommend_discrete(
         from botorch.optim import optimize_acqf_discrete
 
         # determine the next set of points to be tested
-        candidates_tensor = to_tensor(candidates_comp)
+        candidates_comp = subspace_discrete.transform(candidates_exp)
         points, _ = optimize_acqf_discrete(
-            self._botorch_acqf, batch_size, candidates_tensor
+            self._botorch_acqf, batch_size, to_tensor(candidates_comp)
         )
 
         # retrieve the index of the points from the input dataframe
         # IMPROVE: The merging procedure is conceptually similar to what
         #   `SearchSpace._match_measurement_with_searchspace_indices` does, though using
         #   a simpler matching logic. When refactoring the SearchSpace class to
         #   handle continuous parameters, a corresponding utility could be extracted.
+        # IMPROVE: Maintain order of recommendations (currently lost during merge)
```

Comment:
Yeah, probably this can be improved by changing the merge as you say. My bigger concern is misuse the mechanism, and there are at least two possible ways:
* The first one is the one you described, i.e. using a bigger batch size to filter down later. The fundamental problem associated this approach is what we also describe in the docs:
  <img width="1291" alt="image" src="https://github.com/user-attachments/assets/efe7fcde-cee7-4e79-8df2-e42079d99b4c">
* Even if we ignore the above, I think things become problematic when the ratios become extreme. I remember Marko mentioning some use case where the unfiltered list was orders of magnitude larger that what was actually needed, and I don't think botorch's batch mechanism scale well to very large sizes (in terms of quality)
* And finally, the order ONLY applies – if at all – to the *sequential* version, not the batch version. And people might miss this point.

So whatever we do, we need to make the implications very clear to the users. Shall we open an issue for that or how to you want to proceed? I'd like to close this thread here so that I can finish the PR

Reply:
opened #353 

I think as long as the actual feature of batch restrictions is missing, no one cares about the subtleties you mentioned and will use this as a workaround no matter what